client-local/src/main/java/com/cloudera/livy/client/local/LocalClient.java

```diff
@@ -248,76 +251,60 @@ public void run() {
       // mode, the driver options need to be passed directly on the command line. Otherwise,
       // SparkSubmit will take care of that for us.
       String master = conf.get("spark.master");
+      LOG.info("Master : " + master);
+      launcher.setMaster(master);
```

Comment:
I see there's a check in the next line to make sure it's not null, so this code should be added after that check.